src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/DefaultRazorSemanticTokensInfoService.cs

```diff
@@ -482,16 +298,7 @@ internal static bool TryGetMinimalCSharpRange(RazorCodeDocument codeDocument, [N
                 previousResult = result;
             }
 
-            var tokensResult = new SemanticTokensResponse(resultId, data.ToArray(), isFinalizedCSharp);
-            return tokensResult;
-        }
-
-        private static async Task<VersionStamp> GetDocumentSemanticVersionAsync(DocumentSnapshot documentSnapshot)
-        {
-            var documentVersionStamp = await documentSnapshot.GetTextVersionAsync();
-            var semanticVersion = documentVersionStamp.GetNewerVersion(documentSnapshot.Project.Version);
-
-            return semanticVersion;
+            return data.ToArray();
```

Comment:
Does the `GetData` method get used by anything else? Seems like this could be separated, perhaps by inlining it? At the moment data is a List, which we fill from an enumerator, then convert to an array. This method could just be the enumerator.

Reply:
Moving the ToImmutableArray call to here makes this make more sense now, thanks :)